new-log-viewer/src/typings/worker.ts

```diff
@@ -64,8 +66,20 @@ type WorkerReqMap = {
         cursor: CursorType,
         decoderOptions?: DecoderOptionsType
     },
+    [WORKER_REQ_CODE.QUERY_LOG]: {
+        searchString: string,
+        isRegex: boolean,
+        isCaseSensitive: boolean,
+    },
 };
 
+type ChunkResultType = {
+    logEventNum: number;
+    message: string;
+    matchRange: [number, number];
+};
+type ChunkResults = Record<number, ChunkResultType[]>;
```

Comment:
Instead of sending back the results as an `Object`, have we considered a `Map`? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

In addition to performance gain and mitigating a few caveats of `Object`s, there is one significant benefit in using a `Map` in our code:

> [A Map's keys can be any value (including functions, objects, or any primitive).](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#:~:text=A%20Map%27s%20keys%20can%20be%20any%20value%20(including%20functions%2C%20objects%2C%20or%20any%20primitive).)
- We don't have to cast the keys from strings into numbers whenever we retrieve those.